packages/react-native/src/primitives/FederatedProviderButton/FederatedProviderButton.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useCallback } from 'react';
+import {
+  PressableStateCallbackType,
+  StyleProp,
+  View,
+  ViewStyle,
+} from 'react-native';
+
+import { Button, Label } from '..';
+
+import { styles } from './styles';
+import { FederatedProviderButtonProps } from './types';
+
+export default function FederatedProviderButton({
+  children,
+  Icon,
+  onPress,
+  style,
+  textStyle,
+  ...rest
+}: FederatedProviderButtonProps): JSX.Element {
+  const containerStyle = useCallback(
+    ({ pressed }: PressableStateCallbackType): StyleProp<ViewStyle> => {
+      const pressedStateStyle =
+        typeof style === 'function' ? style({ pressed }) : style;
+      return [styles.container, pressedStateStyle];
+    },
+    [style]
+  );
+
+  return (
+    <Button {...rest} onPress={onPress} style={containerStyle}>
+      <View style={styles.icon}>{Icon}</View>
```

Comment:
Why does it need to be wrapped in a `View`? Can we just just apply the style to the `Icon`:
```
<Icon style={styles.icon} />
```

Reply:
![IconCode](https://user-images.githubusercontent.com/48109584/197031099-6414e216-1076-4ecd-8828-655d0aef6728.png)

Reply:
Thinking that might be because the prop is `React.ReactElement` and not `React.ComponentType`?

Reply:
That fixes it, but then the `Icon` component complains that it's missing the `source` prop

Reply:
Maybe we should just pass the ` source` prop like @ioanabrooks suggested